acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+"""Testing spaces and subnets settings for app deployment to lxd container.
+   This is an validation of Bug 1685782:
+   https://bugs.launchpad.net/juju/+bug/1685782
+   The test procedures are based on comment #29.
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
+from deploy_stack import BootstrapManager
+
+from utility import (
+    JujuAssertionError,
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+log = logging.getLogger("assess_lxd_space")
+
+
+def assert_initial_spaces(client):
+    """Initial spaces status after bootstrap should show as:
```

Comment:
I don't think this is universally true; do we intend to extend these tests beyond AWS? The MAAS provider supports space discovery, and I *think* that it will be populated just after bootstrap. You should check that because I'm not an expert on this, but I think you'll need to at least handle MAAS specially.